pkg/operator/resource_reconciler.go

```diff
@@ -457,3 +471,14 @@ func ListMatchingNamespaces(selector labels.Selector, nsInf cache.SharedIndexInf
 	}
 	return ns, nil
 }
+
+// isOwnershipWrong returns false if the controller is the "owner" of the prometheus
+// or ownership has not been setted yet
+// Whether it's owner is determined by the value of label 'controllerID'
+// annotation. If the value matches the controllerID then it owns it.
+func (rr *ResourceReconciler) isOwnershipWrong(labels map[string]string) bool {
```

Comment:
I think it's more readable to use positive assertions: `isOwner` or `hasOwnership`

And we also need to consider the case where the the operator doesn't set `-controller-id`, in that case it shouldn't touch CRs that have a `controllerID` annotation. 

of course, if the operator doesn't set `-controller-id`, it should take care of CRs without `controllerID` annotation...

Reply:
My idea is to force set a rand controller id, because if not, we can have exactly the same issue than before. 
hasOwnership() is not valid because if could be false and we need to enqueue the object in any case.
There is two scenarios:
- label is empty, and we need to set up
- label is correct and match operator controller id.

Reply:
I'm afraid a "rand controller id" would be more complex: we'll need to persist that on restarts and not knowing the ID in advance would make configuring the CRs to be managed by that operator more difficult...

I think that well chosen controller ID can prevent collisions.

The same overlap may happen with Service->Pods e.g. I can easily expose a Pod to 2 service by mistake  but that doesn't prevent users to chose good labels to prevent that...

Reply:
I assume that what Mario wants to point is the case where you may find out that there's more than one operator without controller-id, which would lead us to the initial situation again.

Reply:
If we implement this:

```
And we also need to consider the case where the the operator doesn't set -controller-id, in that case it shouldn't touch CRs that have a controllerID annotation.
```

users will have a way to "isolate/protect" their CRs.

Reply:
The initial plan wasn't to solve all edge cases: if 2 operators start without controller ids then we're back to the original problem as they might fight for the same resource.

The main goal is that we can "bind" objects to a specific operator when multiple operators run on the same cluster. It's the responsibility of the person deploying the Prometheus object to annotate it with the expected controller id. In the case of OCP, it means that we will start the openshift-monitoring operator with `--controller-id=foo` and we will annotate the k8sPrometheus object with `operator.prometheus.io/controller-id: foo`.